src/cmd/cmd_replay.ml

```diff
@@ -106,8 +109,27 @@ let run_file ~unsafe ~optimize ~entry_point ~invoke_with_symbols filename model
 
     let in_replay_mode () = Ok 1l
 
-    let label _ id _ =
-      Logs.info (fun m -> m "reached label %ld@." id);
+    let rec make_str m accu i =
+      let open Concrete_choice in
+      let* p = Concrete_memory.load_8_u m i in
+      if Int32.gt p 255l || Int32.lt p 0l then trap `Invalid_character_in_memory
+      else
+        let ch = char_of_int (Int32.to_int p) in
+        if Char.equal ch '\x00' then return (List.rev accu |> Array.of_list)
+        else make_str m (ch :: accu) (Int32.add i (Int32.of_int 1))
+
+    let cov_label_is_covered id =
+      let open Concrete_choice in
+      let* id = select_i32 id in
+      return @@ Value.const_i32
+      @@ if Hashtbl.mem covered_labels id then 1l else 0l
+
+    let cov_label_set m id str_ptr =
+      let* chars = make_str m [] str_ptr in
+      let* id = Concrete_choice.select_i32 id in
+      let str = String.init (Array.length chars) (Array.get chars) in
+      Hashtbl.add covered_labels id str;
+      Fmt.pr "reached %ld@." id;
```

Comment:
Can you use the new `Logs` module please? You have pointers over here: https://ocamlpro.github.io/owi/hacking/guidelines.html#printing